### ISSUE TITLE
Feat: Support show annotations

### ISSUE THREAD

**Issue by zackzhangkai** [kubernetes/kubectl#1010]
**What would you like to be added**:

support `kubectl get pod/depoloyment/svc xxx --show-annotations`

**Why is this needed**:

Now, we have `kubectl get pod --show-labels`. I think the annotations should also be supported to get directly by `kubectl`.

**Comment by eddiezane** [comment]
@soltysh to take a look about current implementation.

**Comment by FalcoSuessgott** [comment]
I studied the implementation of printing out `kubectl get pods --show-labels` and I would like to give it a try with `--show-annotations`.

May I?

**Comment by soltysh** [comment]
I'm not convinced of the value of adding yet another flag to `kubectl get`. We already have a problem with too many flags per each command which makes it very hard for new-comers and even old-timers to graps all of them. This in particular brings the additional complexity which comes from where the implementation would reside, and that's inside a shared library.

**Comment by FalcoSuessgott** [comment]
Hi @soltysh,

I now got a tested and working version of displaying the annotations, the code is quite redundant but I faced 2 issues as I will explain later: 

```bash
./kubectl get po --show-annotations
NAME    READY   STATUS    RESTARTS   AGE    ANNOTATIONS
nginx   1/1     Running   0          6d1h   cni.projectcalico.org/podIP=10.233.70.3/32,cni.projectcalico.org/podIPs=10.233.70.3/32
```

As you said, it easily gets too complex:

1. the `LABELS` column is implemented as always the last column, because the labels are usually the longest, now with the `ANNOTATIONS` column we would have to find a way how to display the information in a nicely matter, without getting too long:

```bash
/kubectl get po --show-annotations --show-labels
NAME    READY   STATUS    RESTARTS   AGE    LABELS              ANNOTATIONS
nginx   1/1     Running   0          6d1h   run=nginx,test=ok   cni.projectcalico.org/podIP=10.233.70.3/32,cni.projectcalico.org/podIPs=10.233.70.3/32
```

2. With the `--column-label` we can filter specific labels and display them in their own column. Now should we implement the same functionality for annotations (e.g `--column-annotations`), it then gets easily complex with some edge cases and maybe more flags.

I thought about combining annotations and labels together in `metadata` column and indicating them with a `annotation/` prefix but this was just an idea.

My 2 cents of my approach..

Kind regards,

Tom
